spec/requests/health_controller_spec.rb

```diff
@@ -0,0 +1,42 @@
+require 'rails_helper'
+
+RSpec.describe 'Health Controller', type: :request do
+  describe 'GET #show' do
+    context 'when the application is up' do
+      it 'returns a successful response with HTML' do
+        get '/'
+        expect(response).to have_http_status(:ok)
+        expect(response.content_type).to eq('text/html; charset=utf-8')
+        expect(response.body).to include('<html><body style="background-color: green"></body></html>')
+      end
+
+      it 'returns a successful response with JSON' do
+        get '/', headers: { 'ACCEPT' => 'application/json' }
+        expect(response).to have_http_status(:ok)
+        expect(response.content_type).to eq('application/json; charset=utf-8')
+        expect(response.body).to eq({ message: 'OK' }.to_json)
+      end
+    end
+
+    context 'when the application is down' do
+      before(:each) do
```

Comment:
Usage of `before` blocks in Rspec should be avoided: https://nimblehq.co/compass/development/code-conventions/ruby/rspec/#convention-15

There is a lot of complexity under the hood